src/zmalloc.c

```diff
@@ -666,6 +708,21 @@ size_t zmalloc_get_frag_smallbins(void) {
         sz = sizeof(size_t);
         assert(!je_mallctl(buf, &curslabs, &sz, NULL, 0));
 
+        /* When lua uses jemalloc arena, we need to remove it from frag. */
+        if (lua_arena != UINT_MAX) {
+            size_t lua_curregs;
+            snprintf(buf, sizeof(buf), "stats.arenas.%u.bins.%d.curregs", lua_arena, j);
+            sz = sizeof(size_t);
+            assert(!je_mallctl(buf, &lua_curregs, &sz, NULL, 0));
+            curregs = curregs - lua_curregs;
+
+            size_t lua_curslabs;
+            snprintf(buf, sizeof(buf), "stats.arenas.%u.bins.%d.curslabs", lua_arena, j);
+            sz = sizeof(size_t);
+            assert(!je_mallctl(buf, &lua_curslabs, &sz, NULL, 0));
+            curslabs = curslabs - lua_curslabs;
+        }
```

Comment:
not sure if it would be better to add a new argument `unsigned int arena` to `zmalloc_get_frag_smallbins`,
then substract all small frag from the lua small frag.
although it may double the loop, but the code will look cleaner.

Reply:
i've split them up now